ext/ddtrace_profiling_native_extension/stack_recorder.c

```diff
@@ -330,15 +332,15 @@ static void stack_recorder_typed_data_free(void *state_ptr) {
   ruby_xfree(state);
 }
 
-static VALUE _native_initialize(DDTRACE_UNUSED VALUE _self, VALUE recorder_instance, VALUE cpu_time_enabled, VALUE alloc_samples_enabled, VALUE heap_samples_enabled) {
+static VALUE _native_initialize(DDTRACE_UNUSED VALUE _self, VALUE recorder_instance, VALUE cpu_time_enabled, VALUE alloc_samples_enabled, VALUE heap_samples_enabled, VALUE heap_size_enabled) {
   ENFORCE_BOOLEAN(cpu_time_enabled);
   ENFORCE_BOOLEAN(alloc_samples_enabled);
   ENFORCE_BOOLEAN(heap_samples_enabled);
 
   struct stack_recorder_state *state;
   TypedData_Get_Struct(recorder_instance, struct stack_recorder_state, &stack_recorder_typed_data, state);
 
-  state->heap_recorder = heap_recorder_init();
+  state->heap_recorder = heap_recorder_init(heap_size_enabled == Qtrue);
```

Comment:
Minor: Don't forget the `ENFORCE_BOOLEAN`, to avoid typing accidents ;)

Also, perhaps we should raise if `heap_samples_enabled == false` but `heap_size_enabled == true`